packages/api/amplify_api/example/lib/rest_api_view.dart

```diff
@@ -79,16 +75,16 @@ class _RestApiViewState extends State<RestApiView> {
 
   void onGetPressed() async {
     try {
-      RestOperation restOperation = Amplify.API.get(
-          restOptions: RestOptions(
-        path: _apiPathController.text,
-      ));
+      final restOperation = Amplify.API.get(
+        _apiPathController.text,
+      );
 
       _lastRestOperation = restOperation;
-      RestResponse response = await restOperation.response;
+      final response = await restOperation.value;
 
       print('Get SUCCESS');
-      print(response);
+      print(response.statusCode);
+      print(await response.decodeBody());
```

Comment:
Having code similar to the following all over the place seems tedious.

```dart
final response = await restOperation.value;
final body = await response.decodeBody();
```

I would assume most customers would just want to decoded body. Maybe we extend `CancelableOperation` to make the common things customers would do less tedious. I had included the following extension in a comment above discussing the deprecation of `.response`. Something similar is what I was thinking.

```dart
extension CancelableOperationX on CancelableOperation<AWSStreamedHttpResponse> {
  Future<String> get response async {
    final httpResponse = await value;
    return httpResponse.decodeBody();
  }
}
```

I would need to spend quite a bit more time playing around with these changes to determine what would be most useful, so if you don't think this adds value, I trust your judgement here more than mine.

Reply:
Spoke offline about this but I think good thing to consider for API review as changes here still pending that.

Reply:
Agree we need something more fluent. Jordan's suggestion is pretty nice and saves a deprecation/new API

Reply:
revised this based on last convo, open to further suggestions